synthesizer/src/vm/helpers/sequential_op.rs

```diff
@@ -0,0 +1,133 @@
+// Copyright (c) 2019-2025 Provable Inc.
+// This file is part of the snarkVM library.
+
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at:
+
+// http://www.apache.org/licenses/LICENSE-2.0
+
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+use crate::vm::*;
+use console::network::prelude::Network;
+
+use std::{fmt, thread};
+use tokio::sync::oneshot;
+
+impl<N: Network, C: ConsensusStorage<N>> VM<N, C> {
+    /// Launches a thread dedicated to the sequential processing of storage-related
+    /// operations.
+    pub fn start_sequential_queue(
+        &self,
+        request_rx: mpsc::Receiver<SequentialOperationRequest<N>>,
+    ) -> thread::JoinHandle<()> {
+        // Spawn a dedicated thread.
+        let vm = self.clone();
+        thread::spawn(move || {
```

Comment:
Do you know if this code is run in WASM? I am unsure if you can spawn threads there.

Reply:
Thanks for thinking of us Kai! We don't use `VM` object in wasm :). Most `wasm` usage is strictly for execution or using individual pieces of tooling in SnarkVM so I wouldn't worry about the `wasm` target at all if you're dealing with changes that ONLY affect the `VM` object.